test/e2e/data/infrastructure-docker/main/clusterclass-quick-start.yaml

```diff
@@ -54,6 +54,24 @@ spec:
         - type: e2e.remediation.condition
           status: "False"
           timeout: 20s
+    machinePools:
+    - class: default-worker
+      template:
+        metadata:
+          labels:
+            ClusterClass.machinePool.label: "ClusterClass.machinePool.labelValue"
+          annotations:
+            ClusterClass.machinePool.annotation: "ClusterClass.machinePool.annotationValue"
+        bootstrap:
+          ref:
+            apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+            kind: KubeadmConfigTemplate
+            name: quick-start-default-worker-bootstraptemplate
```

Comment:
Let's use a different bootstraptemplate here. Let's have quick-start-md-default-worker-bootstraptemplate and quick-start-mp-default-worker-bootstraptemplate

And then put the BootstrapTemplate labels/annotations only where they belong. Otherwise we add them to both MD and MP Machines and it becomes weirdly mixed up :)